examples/browser/Dockerfile

```diff
@@ -1,43 +1,26 @@
-# How to use this image
-# 
-# The following instructions assume that you do not have Docker or Node
-# installed on your computer. Instead, you will use the online Docker
-# playground website 'Play With Docker (PWD)' to run the Theia browser example
-# by just using PWD through your browser.
-# 
-# 1. First, open up http://labs.play-with-docker.com/ in your browser and
-#    create a Play With Docker session and add a new instance to your Docker
-#    playground.
-# 
-# 2. Next, you will need to download this Dockerfile into your PWD session with
-#    wget.
-# 
-# $ wget https://raw.githubusercontent.com/theia-ide/theia/master/examples/browser/Dockerfile
-# 
-# 3. Next, ask Docker to build the image. This will take some time.
-# 
-# $ docker build -t theia .
-# 
-# 4. Now use Docker to run the image you just built.
-# 
-# $ docker run -d -p 0.0.0.0:3000:3000 theia
-# 
-# 5. There should now be a 3000 link at the top of your PWD window. Click on it
-#    to try out the Theia browser example!
-#    
+# Build image:
+#    $ docker build --build-arg GITHUB_TOKEN=<token> -t theia .
+#
+#    Note to build that image you may need to build theia-dev image using Dockerfile.dev
+#
+#    Note that sometimes you can reach Github API rate limit, so to solve this
+#    issue please set build-arg GITHUB_TOKEN as per example
+#
+# Running theia in a docker container:
+#    $ docker run -d -p 0.0.0.0:3000:3000 theia
+#
+#    To access theia-ide open localhost:3000 in your browser
+#
 #    Note that you may get an "Error forwarding request." error after clicking
 #    on the 3000 link. Please wait a few seconds before trying again as Theia
 #    may not have completely finished starting up yet.
 #
 # Note: If you want Java language support you will need to extend this image yourself
 # by adding a JDK to the image and making sure it is accessible via the system PATH.
 
-FROM node:8
-RUN useradd --create-home theia
-WORKDIR /home/theia
-RUN rm -rf /opt/yarn && rm -f /usr/local/bin/yarn && rm -f /usr/local/bin/yarnpkg
-RUN apt-get update && apt-get install -y npm && npm install -g yarn@1.3.2
-USER theia
+FROM theia-dev
+ARG GITHUB_TOKEN
+ENV GITHUB_TOKEN ${GITHUB_TOKEN}
```

Comment:
How is the GITHUB_TOKEN being used?

Reply:
Hello @svenefftinge :) 
So it is needed sometimes to build theia, during the build a lot of GH API calls performed by the build system I guess, this happens when many devs working from a single network. In my case theia build was failing due to GH API rate limit. So to solve this GH_TOKEN should be used.

Reply:
Hi :) Could you point me to an example, where the GITHUB_TOKEN env is required during the Theia build? I don't have the variable set locally and don't have issues.

Reply:
@svenefftinge We have seen this, once in a while, while cloning GH repos anonymously. It might be because we work behind a proxy, and from GH's perspective a bunch of us seem to come from the same IP address, and we so hit some type of "per user" transfer limit. Using the token, you are authenticated, and so the limit is not "shared", but only for your own user. 

It's a bit hard to reproduce on demand, however...

Reply:
Thanks. Would be really interesting to understand why that is.

Reply:
some more info: https://developer.github.com/changes/2012-10-14-rate-limit-changes/

Reply:
I;ve just removed token and my build failed:
```
yarn install v1.3.2
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
warning Pattern ["gulp-util@latest"] is trying to unpack in the same destination "/home/theia/.cache/yarn/v1/npm-gulp-util-3.0.8-0054e1e744502e27c04c187c3ecc505dd54bbb4f" as pattern ["gulp-util@^3.0.0","gulp-util@^3.0.1"]. This could result in a non deterministic behavior, skipping.
warning Pattern ["request@latest"] is trying to unpack in the same destination "/home/theia/.cache/yarn/v1/npm-request-2.83.0-ca0b65da02ed62935887808e6f510381034e3356" as pattern ["request@^2.81.0","request@~2.83.0","request@^2.82.0","request@^2.83.0","request@2","request@^2.79.0","request@^2.45.0"]. This could result in a non deterministic behavior, skipping.
warning Pattern ["through@latest"] is trying to unpack in the same destination "/home/theia/.cache/yarn/v1/npm-through-2.3.8-0dd4c9ffaabc357960b1b724115d7e0e86a2e1f5" as pattern ["through@^2.3.6","through@^2.3.4","through@>=2.2.7 <3","through@2"]. This could result in a non deterministic behavior, skipping.
warning vscode-base-languageclient@0.0.1-alpha.3: The engine "vscode" appears to be invalid.
info fsevents@1.1.3: The platform "linux" is incompatible with this module.
info "fsevents@1.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning " > istanbul-instrumenter-loader@3.0.0" has unmet peer dependency "webpack@^2.0.0 || ^3.0.0".
warning "workspace-aggregator-565afa12-12f6-47bb-879e-9494ca3ada55 > @theia/application-package > font-awesome-webpack@0.0.5-beta.2" has unmet peer dependency "font-awesome@>=4.3.0".
[5/5] Building fresh packages...
error /home/theia/node_modules/vscode-ripgrep: Command failed.
Exit code: 1
Command: node ./dist/postinstall.js
Arguments:
Directory: /home/theia/node_modules/vscode-ripgrep
Output:
Renaming _node_modules => node_modules
Downloading ripgrep failed: Error: API rate limit exceeded for 193.254.196.193. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

but to reproduce this I had to wipe cached modules with `git clean -xdf`
So this is related to getting some dependencies I think

Reply:
Ok, I see it is here : https://github.com/roblourens/vscode-ripgrep/blob/master/dist/postinstall.js#L36

Reply:
:) PR build also failed due to this issue https://ci.appveyor.com/project/kittaakos/theia/build/1.0.2035 
```
Renaming _node_modules => node_modules
Downloading ripgrep failed: Error: API rate limit exceeded for 67.225.138.82. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

Reply:
@riuvshin In this specific PR, there are no code changes, so I think it would be safe to ignore the failure. I pinged @kittaakos on gitter, to ask that he restart this build. Let's see if it happens twice in a row.
 
Would you please register a GH issue about this? It's the first time I am aware of this happening on appveyor; we might need to do something about it, if this is more than a one-off.

Reply:
Should not we set `GITHUB_TOKEN` env variable for appveyor as well?

Reply:
I've updated `GITHUB_TOKEN` in travis and appveyor on an encrypted access token.

Reply:
yes, that should help